lib/rubocop/cop/style/invertible_unless_condition.rb

```diff
@@ -88,6 +91,47 @@ def inheritance_check?(node)
             (argument.const_type? && argument.short_name.to_s.upcase != argument.short_name.to_s)
         end
 
+        def preferred_condition(node)
+          case node.type
+          when :begin
+            if node.parenthesized_call?
+              "(#{preferred_condition(node.children.first)})"
+            else # TODO: What about other kinds of begin nodes?
+              preferred_condition(node.children.first)
+            end
+          when :send
+            preferred_send_condition(node)
+          when :or, :and
+            preferred_logical_condition(node)
+          end
+        end
+
+        def preferred_send_condition(node) # rubocop:disable Metrics/MethodLength
+          receiver_source = node.receiver.source
+          return receiver_source if node.method?(:!)
+
+          inverse_method_name = inverse_methods[node.method_name]
+          if node.operator_method?
+            "#{receiver_source} #{inverse_method_name} #{node.first_argument.source}"
+          elsif node.arguments?
+            argument_list = node.arguments.map(&:source).join(', ')
+            if node.parenthesized?
+              "#{receiver_source}.#{inverse_method_name}(#{argument_list})"
+            else
+              "#{receiver_source}.#{inverse_method_name} #{argument_list}"
+            end
+          else
+            "#{receiver_source}.#{inverse_method_name}"
+          end
+        end
```

Comment:
I kept this version of this method as a separate commit, in case we prefer the one which does not introduce an offense:

```ruby
def preferred_send_condition(node)
  receiver_source = node.receiver.source
  return receiver_source if node.method?(:!)

  inverse_method_name = inverse_methods[node.method_name]
  dot_or_space = node.dot? ? '.' : ' '
  args = node.arguments.map(&:source).join(', ')
  args = node.parenthesized? ? "(#{args})" : " #{args}" if node.arguments?

  "#{receiver_source}#{dot_or_space}#{inverse_method_name}#{args}"
end
```

While this approach is shorter, I don't know if it's actually clearer, as it condenses the logic into fewer more complex lines.

Another approach would be to extract more methods, but I feel like that would also hurt the overall readability.

The longer approach makes it simpler to see what the outcome is:

Scenario|Preferred Implementation
-|-
Unary `!`|`receiver_source`
Operator|`"#{receiver_source} #{inverse_method_name} #{node.first_argument.source}"`
Parentheses|`"#{receiver_source}.#{inverse_method_name}(#{argument_list})"`
No Parentheses|`"#{receiver_source}.#{inverse_method_name} #{argument_list}"`
No arguments|`"#{receiver_source}.#{inverse_method_name}"`

Reply:
I managed to put together a similar implementation which doesn't trigger an offense.